File: conformance/bundles/09-expected-message-mismatch/BoomSteps.java
package dev.varar.conformance.bundle09;

import dev.varar.Registrar;
import dev.varar.State;
import dev.varar.StateBinder;
import dev.varar.StepDefinitions;

/**
 * Java sibling of {@code boom.steps.ts} / {@code boom.steps.py} (bundle {@code
 * 09-expected-message-mismatch}). Throws a message that does NOT contain the expected
 * substring "expected message", so the expected-failure is NOT satisfied and the
 * example fails at the trace stage.
 */
public final class BoomSteps implements StepDefinitions {

    record Ctx() implements State {}

    @Override
    public void defineSteps(Registrar registrar) {
        StateBinder<Ctx> s = registrar.steps(Ctx::new);

        s.stimulus(
                "I always boom",
                (Ctx ctx) -> {
                    throw new RuntimeException("actual different error");
                });
    }
}

File: conformance/bundles/09-expected-message-mismatch/boom.steps.kt
@file:JvmName("BoomSteps")

// Kotlin sibling of boom.steps.ts / boom.steps.py / BoomSteps.java (bundle
// 09-expected-message-mismatch): throws a message NOT containing the expected
// substring, so the error fence is not satisfied at the trace stage.
package dev.varar.kotlin.conformance.bundle09

import dev.varar.kotlin.stimulus
import dev.varar.kotlin.steps

class Ctx

val steps = steps(::Ctx) {
    stimulus("I always boom") {
        throw RuntimeException("actual different error")
    }
}
